src/utils/platform.ts

```diff
@@ -1,57 +1,47 @@
-export function isChromiumBased() {
-    return navigator.userAgent.toLowerCase().includes('chrome') || navigator.userAgent.toLowerCase().includes('chromium');
-}
-
-export function isFirefox() {
-    return navigator.userAgent.includes('Firefox');
-}
-
-export function isVivaldi() {
-    return navigator.userAgent.toLowerCase().includes('vivaldi');
-}
-
-export function isYaBrowser() {
-    return navigator.userAgent.toLowerCase().includes('yabrowser');
-}
-
-export function isOpera() {
-    const agent = navigator.userAgent.toLowerCase();
-    return agent.includes('opr') || agent.includes('opera');
-}
-
-export function isEdge() {
-    return navigator.userAgent.includes('Edg');
-}
-
-export function isWindows() {
-    if (typeof navigator === 'undefined') {
-        return null;
-    }
-    return navigator.platform.toLowerCase().startsWith('win');
-}
-
-export function isMacOS() {
-    if (typeof navigator === 'undefined') {
-        return null;
-    }
-    return navigator.platform.toLowerCase().startsWith('mac');
-}
-
-export function isMobile() {
-    if (typeof navigator === 'undefined') {
-        return null;
+let userAgent = typeof navigator === 'undefined' ? 'some useragent' : navigator.userAgent.toLowerCase();
+let platform = typeof navigator === 'undefined' ? 'some platform' : navigator.platform.toLowerCase();
+
+
+export const isChromium = userAgent.includes('chrome') || userAgent.includes('chromium');
+export const isFirefox = userAgent.includes('firefox');
+export const isVivaldi = userAgent.includes('vivaldi');
+export const isYaBrowser = userAgent.includes('yabrowser');
+export const isOpera = userAgent.includes('opr') || userAgent.includes('opera');
+export const isEdge = userAgent.includes('edg');
+export const isWindows = platform.startsWith('win');
+export const isMacOS = platform.startsWith('mac');
+export const isMobile = userAgent.includes('mobile');
+export const isShadowDomSupported = typeof ShadowRoot === 'function';
+export let chromiumVersion: string = null;
+let m = userAgent.match(/chrom[e|ium]\/([^ ]+)/);
+if (m && m[1]) {
+    chromiumVersion = m[1];
+} else {
+    chromiumVersion = '';
+}
+
+// Clean up memory
```

Comment:
This is a micro optimization :) They are rather small strings and Garbage Collector will remove it from memory after some time, because no other code references them.